src/plugins/mer/merdeploysteps.h

```diff
@@ -265,10 +277,31 @@ class MerDeployStepWidget : public ProjectExplorer::BuildStepConfigWidget
     void formatAndSetSummaryText(const QString &summaryText);
 private slots:
     void commandArgumentsLineEditTextEdited();
-private:
+protected:
     Ui::MerDeployStepWidget m_ui;
 };
 
+class MerRpmBuildStepWidget : public MerDeployStepWidget
+{
+    Q_OBJECT
+public:
+    MerRpmBuildStepWidget(MerProcessStep *step, Sfdk::BuildEngine *engine);
+    void setSignPackages(bool sign);
+    void setSignPackageKey(const QString &keyFile);
+    void setSignPackagePass(const QString &passphrasefile);
+signals:
+    void signPackagesChanged(bool sign);
+    void signPackageKeyChanged(const QString &key);
+    void signPackagePassFileChanged(const QString &file);
+private:
+    void initKeyChooser();
+    void initKeyPassphraseChooser();
+    QCheckBox *m_signPackagesCheckBox;
+    QLineEdit *m_signPackageKeyEdit;
+    Utils::PathChooser *m_signPackageKeyPassFileChooser;
+    Sfdk::BuildEngine *m_engine;
+};
+
```

Comment:
ditto, should be implemented in MerBuildConfigurationAspect/Widget instead.

Reply:
For Rsync deploy step, signing is not necessary, but build configuration is common for all methods. Wouldn't that be weird?

Reply:
rsync deployment is rather a hack that needs to be used in cooperation with regular RPM deployment. Anyway it is the "package" step, not the "deploy" step what is controlled by those options, so it's all OK.